### ISSUE TITLE
Data quality profile fields not viewable

### ISSUE THREAD

**Issue by alexhuang091** [AtlasOfLivingAustralia/la-pipelines#392]
Steps to reproduce:

1. Choose data profile `ALA`
2. Click the `<i> `on Exclude absence records which pops up the category detail
![2](https://user-images.githubusercontent.com/61677987/120263952-1b373d80-c2e0-11eb-8c2b-aefd020df31a.jpg)

This is because we send a request like this to get field detail
`http://localhost:8078/index/fields?fl=basis_of_record`

but the returned json is like 

```
[
    {
        "name": "basisOfRecord",
        "jsonName": "basisOfRecord",
        "dwcTerm": "basisOfRecord",
        "downloadName": "basisOfRecord",
        "dataType": "string",
        "classs": "Record",
        "indexed": true,
        "docvalue": true,
        "stored": true,
        "multivalue": false,
        "description": "Record type",
        "downloadDescription": "Basis Of Record",
        "info": "What this is a record of e.g. specimen, human observation, fossil http://rs.tdwg.org/dwc/terms/basisOfRecord",
        "numberDistinctValues": 8
    }
]
```

See the request name is **basis_of_record** but returned name is **basisOfRecord**

which makes below function in` search.js` fail because 
```
$('.DQCategoryDetailsLink').click(function() {

    $.each(keys, function (index, key) {
        if (map.has(key)) { // this will never be true in currrent enviroment, need to be fixed!
            html += "<tr><td style='word-break: normal'>" + key + "</td><td style='word-break: break-word'>" + replaceURL(map.get(key)[0]) + "</td><td style='word-break: normal'>" + replaceURL(map.get(key)[1], 'Link') + "</td></tr>";
        }
})

```

So in search.js we need a mapping to map the old name to new name

**Comment by brucehyslop** [comment]
I think a better solution is to update the profile filter config. 

I've created new profile on https://data-quality-service-test.ala.org.au which is a clone of the `ALA General` with the filters modified to use the new field names and field vocabulary. The profile is called `ALA General - pipelines` and when selected show the same results as the `ALA general` 

![Screen Shot 2021-06-02 at 1 23 33 pm](https://user-images.githubusercontent.com/12233783/120419919-af6bd800-c3a6-11eb-9fe4-e4133adcb6b7.png)

![Screen Shot 2021-06-02 at 1 27 09 pm](https://user-images.githubusercontent.com/12233783/120419963-c0b4e480-c3a6-11eb-8860-a5ce942ce912.png)

**Comment by brucehyslop** [comment]
I've updated the prod data quality profiles and replaced them on https://data-quality-service-test.ala.org.au

The previously enabled profiles have been renamed with the suffix `-old` and "(pre-pipelines).

**Comment by brucehyslop** [comment]
@M-Nicholls I've updated the data quality profiles on test with filters changed to use the new field names and field value vocabulary (for `assertions`, `basisOfRecord`, `duplicateStatus` & `occurrenceStatus`).

**Comment by nickdos** [comment]
This looks very close to prod. Only aspect that is different, is the test site is missing the link to the GH wiki page for detailed info on the assertion details.

![image](https://user-images.githubusercontent.com/532845/121277870-d715ef80-c914-11eb-8a99-f8c85f5ec851.png)

Maybe this can be dealt with a separate issue for wiki content (might already exists?) for a subsequent release? 

@javier-molina feel free to move to done, if above is OK with you.

**Comment by RobinaSanderson** [comment]
Hi @nickdos @javier-molina @brucehyslop 

I think the issue Nick has found above is that the assertions without links are either new or don't have an equivalent in the pipelines.  Also the filters mapping to the wiki pages appear broken. Looking at the ALA General profile example below I've looked at each filter without a link. 
![image](https://user-images.githubusercontent.com/23368496/121316385-9f2a9e80-c94c-11eb-838a-882312b8aa11.png) 

I think the link mapping is held in the messages properties file in Github as per https://github.com/AtlasOfLivingAustralia/DataQuality/issues/116 I'm sorry but I'm not sure how to update this.

The new spatiallyValid filter description could point to the new KB article on spatial validity when it's published.

The following filters are based on assertions using camel case so I assume they do not map to anything in pipelines and will be removed:
- nameNotRecognised
- homonymIssue
- occCultivatedEscapee
- zeroLatitude
- zeroLongitude

There are also the following filters which could link to pages
duplicateStatus:"ASSOCIATED" - https://github.com/AtlasOfLivingAustralia/ala-dataquality/wiki/duplicate_status
coordinateUncertaintyInMeters:[10001 TO *] - https://github.com/AtlasOfLivingAustralia/ala-dataquality/wiki/coordinate_uncertainty
userAssertions:50001 - https://github.com/AtlasOfLivingAustralia/ala-dataquality/wiki/user_assertions
userAssertions:50005 - same as above
outlierLayerCount:[3 TO *] - https://github.com/AtlasOfLivingAustralia/ala-dataquality/wiki/outlier_layer_count
occurrenceStatus:ABSENT - https://github.com/AtlasOfLivingAustralia/ala-dataquality/wiki/occurrence_status
year:[* TO 1700] - https://github.com/AtlasOfLivingAustralia/ala-dataquality/wiki/year

Not sure if basisOfRecord is still being used? If so, these filters map to https://github.com/AtlasOfLivingAustralia/ala-dataquality/wiki/basis_of_record

Maybe this should be moved to it's own issue and I can go through the other profiles?

**Comment by alexhuang091** [comment]
the wiki links should come from messages.properties file and seems Bruce has updated them
```
wiki.assertions=https://github.com/AtlasOfLivingAustralia/ala-dataquality/wiki/assertions
wiki.basisOfRecord=https://github.com/AtlasOfLivingAustralia/ala-dataquality/wiki/basis_of_record
wiki.coordinateUncertaintyInMeters=https://github.com/AtlasOfLivingAustralia/ala-dataquality/wiki/coordinate_uncertainty
wiki.decade=https://github.com/AtlasOfLivingAustralia/ala-dataquality/wiki/occurrence_decade_i
wiki.duplicateStatus=https://github.com/AtlasOfLivingAustralia/ala-dataquality/wiki/duplicate_status
wiki.isDuplicateOf=https://github.com/AtlasOfLivingAustralia/ala-dataquality/wiki/duplicate_record
wiki.license=https://github.com/AtlasOfLivingAustralia/ala-dataquality/wiki/license
wiki.occurrenceStatus=https://github.com/AtlasOfLivingAustralia/ala-dataquality/wiki/occurrence_status
wiki.outlierLayerCount=https://github.com/AtlasOfLivingAustralia/ala-dataquality/wiki/outlier_layer_count
wiki.spatiallyValid=https://github.com/AtlasOfLivingAustralia/ala-dataquality/wiki/geospatial_kosher
wiki.taxonomicIssues=https://github.com/AtlasOfLivingAustralia/ala-dataquality/wiki/taxonomic_issue
wiki.userAssertions=https://github.com/AtlasOfLivingAustralia/ala-dataquality/wiki/user_assertions
wiki.year=https://github.com/AtlasOfLivingAustralia/ala-dataquality/wiki/year
```

on develop branch
https://github.com/AtlasOfLivingAustralia/biocache-service/blob/cf41784f1f57e39f7af7232ebd360ee966c34f6d/src/main/java/au/org/ala/biocache/dao/SearchDAOImpl.java#L3669

infoUrl is set and returned when user calls `ws/index/fields?fl=occurrenceStatus` to get the details of a field. 

**But this function seems to be removed in pipeline code so `https://biocache-test.ala.org.au/ws/index/fields?fl=occurrenceStatus` doesn't return a infoUrl field.**

hubs uses infoUrl as the target of link. No infoUrl then there's no link.

I'll keep an eye on it and when I finish the index issue I can work on this.

**Comment by alexhuang091** [comment]
I mean the link comes from `infoUrl` in ws/index/fields?fl=occurrenceStatus, but in pipeline this field is not returned.

**Comment by javier-molina** [comment]
Help links will be addressed on #419, relevant conversation has been captured in there. 

I will close this now.
